### PR TITLE
GG-33818 Agilent SQL Indexes B+Tree corruption

### PR DESCRIPTION
**Pull request by korlov42**

### CODE REVIEW

**Comment by tledkov-gridgain** [comment]
LGTM

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java [hunk 1]
```diff
@@ -597,53 +601,96 @@ public void copyMetaInfo(MetaPageInfo info) throws IgniteCheckedException {
...
+
                     lastIdxUsed++;
 
-                    if (c != 0)
-                        return fixSort(c, inlineCols[i].sortType);
+                    if (inlinedCmpRes != 0) {
+                        if (!resTrusted)
+                            break;
+
+                        return fixSort(inlinedCmpRes, inlineCols[i].sortType);
```
Comment (ygerzhedovich) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java:632]:
```suggestion
                    if (inlinedCmpRes != 0) {
                        inlinedCmpRes = fixSort(inlinedCmpRes, inlineCols[i].sortType);
                        if (!resTrusted)
                            break;

                        return inlinedCmpRes;
```

File: modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java [hunk 2]
```diff
@@ -597,53 +601,96 @@ public void copyMetaInfo(MetaPageInfo info) throws IgniteCheckedException {
...
+                        dataRowCmpRes = compareValues(v1, v2);
 
-                    if (c != 0)
-                        return fixSort(c, col.sortType);
+                        if (dataRowCmpRes != 0)
+                            return fixSort(dataRowCmpRes, col.sortType);
+                    }
+                }
+                finally {
+                    if (!resTrusted && Integer.signum(inlinedCmpRes) != Integer.signum(dataRowCmpRes)) {
```
Comment (ygerzhedovich) [inline modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java:674]: inlinedCmpRes could have incorrect sign here. Suggestion above should fix the issue.

Comment (korlov42) [reply modules/indexing/src/main/java/org/apache/ignite/internal/processors/query/h2/database/H2Tree.java]: it's OK cause `dataRowCmpRes` doesn't respect the sorting order either

**Comment by tledkov-gridgain** [comment]
LGTM

**Pull request merged**